### PR TITLE
Task/tileserv pipeline

### PR DESCRIPTION
**Pull request by conbrad**
- Adds tileserv infrastructure as part of deployment pipeline
- `nginx` cache and `pg_tileserv` are deployed together as a deployment config
    - `nginx` is built from scratch to enable run permissions
    - `pg_tileserv` is pulled in and and cached in artifactory
    - The `postgresql` cluster is deployed via crunchy which abstracts away a lot of patroni: https://access.crunchydata.com/documentation/postgres-operator/v5/tutorial/getting-started/
 - migrations are run in an init-container before deployment of the above

Notes:
- Sonarcloud complains about duplication in migrations and models, which we can't avoid since definitions need to be unique
-  The `tileserv` module lives on its own outside of our current `api/app` root module since it includes it's own migrations and I think we should move to a more per app module layout so we can selectively deploy applications in the future by leveraging pants: https://www.pantsbuild.org/

# Test Links:
[Percentile Calculator](https://wps-pr-2411.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-2411.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2411.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2411.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2411.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2411.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2411.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-2411.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -132,6 +149,27 @@ jobs:
...
+    steps:
+      - name: Set Variables
+        shell: bash
+        run: |
+          echo "SUFFIX=pr-${{ github.event.number }}" >> $GITHUB_ENV
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: NATS Message Queue
```
Comment (conbrad) [inline .github/workflows/deployment.yml:167]: Change name

File: tileserv/.env [hunk 2]
```diff
@@ -0,0 +1 @@
+TILESERV_WRITE_HOST=localhost
```
Comment (conbrad) [inline tileserv/.env:1]: Delete file

File: tileserv/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+authors = ["Conor Brady <con.brad@gmail.com>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.10"
+sqlalchemy = "^1.4.44"
+alembic = "^1.8.1"
+psycopg2-binary = "^2.9.5"
+python-decouple = "^3.6"
+fire = "^0.4.0"
```
Comment (conbrad) [inline tileserv/pyproject.toml:14]: Delete

File: tileserv/db.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+write_user = config.get('TILESERV_DB_WRITE_USER', 'tileserv')
+postgres_password = config.get('TILESERV_DB_PASSWORD', 'tileserv')
+postgres_write_host = config.get('TILESERV_WRITE_HOST', 'host.docker.internal')
+postgres_port = config.get('TILESERV_DB_PORT', '5432')
+postgres_database = config.get('TILESERV_DB', 'tileserv')
+postgres_url = config.get('DATABASE_URL')
+
+# pylint: disable=line-too-long
+DB_WRITE_STRING = f'postgresql://{write_user}:{postgres_password}@{postgres_write_host}:{postgres_port}/{postgres_database}'
```
Comment (conbrad) [inline tileserv/db.py:13]: Delete

**Review by andrea-williams — APPROVED** [review]

**Review by dgboss — APPROVED** [review]

File: tileserv/.env (hunk 2 above)
[the remark this answers is not in the document]
Comment (conbrad) [reply tileserv/.env]: Thanks, removed

**Pull request merged**